frontend/src/components/Shared/Map/CrossingMap.js

```diff
@@ -232,7 +181,25 @@ class CrossingMap extends React.Component {
 
   onMapClick = e => {
     const { map } = this.state;
-    const features = map.queryRenderedFeatures(e.point);
+    const { showOpen, showClosed, showCaution, showLongterm } = this.props;
+
+    const width = 10;
+    const height = 10;
+    let layersToQuery = [];
+    if (showOpen) layersToQuery.push('openCrossings');
+    if (showClosed) layersToQuery.push('closedCrossings');
+    if (showCaution) layersToQuery.push('cautionCrossings');
+    if (showLongterm) layersToQuery.push('longtermCrossings');
+
+    const features = map.queryRenderedFeatures(
+      [
+        [e.point.x - width / 2, e.point.y - height / 2],
+        [e.point.x + width / 2, e.point.y + height / 2],
+      ],
+      { layers: layersToQuery },
+    );
+
+    // const features = map.queryRenderedFeatures(e.point);
 
     if (features && features[0] && features[0].properties.crossingId) {
```

Comment:
This isn't from this PR, but FYI you could make this a little more succint with:

```
has(`features[0].properties.crossingId`)
```

https://lodash.com/docs#has